modules/Core/hooks/MentionsHook.php

```diff
@@ -18,74 +18,76 @@ class MentionsHook extends HookBase {
-                $params['user']->data()->id,
-                $params['content'],
-                $params['alert_url'],
-                $params['mention_notification_type'],
-                $params['mention_notification_title'],
-            );
+    public static function preCreate(AbstractEvent $event): void {
+        if (!empty($event->content) && isset($event->user)) {
+            if (isset($event->alert_url, $event->mention_notification_type, $event->mention_notification_title)) {
+                $event->content = MentionsParser::parseAndNotify(
+                    $event->user->data()->id,
+                    $event->content,
+                    $event->alert_url,
+                    $event->mention_notification_type,
+                    $event->mention_notification_title
+                );
+            } else {
+                $event->content = MentionsParser::parse(
+                    $event->user->data()->id,
+                    $event->content
+                );
+            }
         }
-
-        return $params;
     }
 
     /**
      * Called before content is edited in the database.
      * - Replaces @mentions with [user] tags.
      */
-    public static function preEdit(array $params = []): array {
-        if (self::validate($params)) {
-            $params['content'] = MentionsParser::parse(
-                $params['user']->data()->id,
-                $params['content'],
+    public static function preEdit(AbstractEvent $event): void {
+        if (!empty($event->content) && isset($event->user)) {
+            $event->content = MentionsParser::parse(
+                $event->user->data()->id,
+                $event->content
             );
         }
-
-        return $params;
     }
 
     /**
      * Parses the [user] tags in a post and replaces them with a link to the user's profile.
      * e.g. [user]1[/user] would instead become <a href="profile/username">@username</a>
      *
-     * @param array $params
-     * @return array
+     * @param AbstractEvent $event
      */
-    public static function parsePost(array $params = []): array {
-        if (parent::validateParams($params, ['content'])) {
-            $params['content'] = self::processUserTags(
-                $params['content'],
+    public static function parsePost(AbstractEvent $event): void {
+        if (!empty($event->content)) {
+             $event->content = self::processUserTags(
+                $event->content,
                 static function (array $userData) {
                     [$userId, $userStyle, $userNickname, $userProfileUrl] = $userData;
                     return '<a href="' . $userProfileUrl . '" data-poload="' . URL::build('/queries/user/', 'id=' . $userId) . '" class="user-mention" style="' . $userStyle . '">@' . Output::getClean($userNickname) . '</a>';
                 }
             );
         }
+    }
 
-        return $params;
+    public static function stripPost(AbstractEvent $event): void {
```

Comment:
hmm, this feels like a smell to me... can we not use the existing `stripContent`?

Reply:
The Text::embedSafe function also used StripPost so i needed to make seperate function that returns string, while the one you tagged is using the event system